logic.js

```diff
@@ -0,0 +1,116 @@
+const Tweeter = function () {
+
+    const _posts = []
+        
+    let postIdCounter = 0
+    let commentIdCounter = 0
+
+    const getPosts = function () {
+        return _posts
+    }
+
+    const addPost = function (text) {
+        if(text!="") {
+        postIdCounter++
+        const newPost = {
+            id: "p" + postIdCounter,
+            text: text,
+            comments: [] 
+        }
+        _posts.push(newPost)
+    }
+    }
+
+    const removePost = function (postID) {
+        let eleIndex = -1
+        for ( i in _posts) {
+            if(_posts[i].id==postID) eleIndex=i 
+        }
+        if(eleIndex!=-1) {
+            _posts.splice(eleIndex,1)
+        }
+    }
+
+    const addComment = function (text,postID) {
+        if(text!="") {
+        commentIdCounter++
+        let newComment = {
+            text:text,
+            id: "c" + commentIdCounter
+        }
+        for(post of _posts) {
```

Comment:
When declaring a new variable (`post`) you should declare it with  _let_, otherwise, the variable will be defined as "_var_"